controllers/drplacementcontrol_controller.go

```diff
@@ -1502,20 +1512,32 @@ func (d *DRPCInstance) hasAlreadySwitchedOver(targetCluster string) bool {
 	return false
 }
 
-func (d *DRPCInstance) isCurrentHomeClusterReadyForRelocation() bool {
+//nolint:cyclop
+func (d *DRPCInstance) isCurrentHomeClusterReadyForRelocation(targetCluster string) bool {
 	const ready = true
+	// This should not happen for relocation
+	if len(d.vrgs) == 0 {
+		d.log.Info("VRG count is 0. Should not happen for relocation")
 
-	homeCluster := d.findPrimaryCluster()
-	// If no cluster is a primary, then we will assume that the cluster(s) is ready for relocation
-	if homeCluster == "" {
-		// we will assume that len(d.vrgs) is greater than 0
-		return true
+		return !ready
+	}
+	// IF len is greater than 1, then the only possibility for this to happen is
+	// when the targetCluster has already been moved to primary. We must insure
+	// that condition is true
+	if len(d.vrgs) > 1 {
+		return d.findPrimaryCluster() == targetCluster
```

Comment:
Hmmm, it is definitely complex to reason about.  But let me first try an easy case:
1. Initial deployment to `East` (one VRG primary)
2. East failed, and we failed over to `West` (two VRGs; both of them are `primary`, but the ManifestWork for `East` is pending a change to `secondary`. Once `East` is up, VRG will change to secondary and at some point, it is deleted).
3. Let's assume after a while, `East` is up and VRG is deleted.
4. The user changes the action to Relocate (at this time we would only have one VRG because of step 3. And line 1527 will not be true. Then the subsequent code will check the condition for relocation readiness.

Let see now little bit more complex case:
1. Initial deployment to `East` (one VRG primary)
2. East failed, and we failed over to `West` (two VRGs; both of them are `primary`, but the ManifestWork for `East` is pending a change to `secondary`. Once `East` is up, VRG will change to secondary and at some point, it is deleted).
3. Let's now assume that `PeerReady` condition is NOT true.
4. The user changes the action to Relocate, then we fail it because `PeerReady` condition is not true.

Let see a more complex case:
1. Initial deployment to `East` (one VRG primary)
2. East failed, and we failed over to `West` (two VRGs; both of them are `primary`, but the ManifestWork for `East` is pending a change to `secondary`. Once `East` is up, VRG will change to secondary and at some point, it is deleted).
3. Let's assume that `PeerReady` condition is true.
4. The user changes the action to Relocate to `East`. At this time, we should only have ONE VRG, which is `primary` (per step 3 assumption).
5. The first call to `isCurrentHomeClusterReadyForRelocation` will ensure that the current home cluster, which is the `primary` VRG is ready for relocation
6. Let's assume it is ready to relocate, the next step is to modify the only VRG we have to `secondary` (that's in `setupRelocation` function)
7. The second call to `isCurrentHomeClusterReadyForRelocation` comes after we modified the VRG for the current home cluster to `secondary` (step 6). Again, at this point, we have only ONE VRG, and we will proceed to executed relocation
8. If we come back here (because relocation is not finished yet) then that statement will be true (we will have TWO VRGs, one `secondary` (old home) and another one `primary` (new home).  And all we have to do is return true because we have a `primary`.

Scary I know :)